bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApiFactory.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.senechome.internal;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenecHomeApiFactory} provides http client based instances to access the web api from senec devices
+ *
+ * @author Steven Schwarznau - Initial contribution
+ */
+@Component(service = SenecHomeApiFactory.class)
+public class SenecHomeApiFactory {
```

Comment:
Please annotate class with `@NonNullByDefault` and adjust the other annotations as necessare (e.g.remove `@NonNull`)

Reply:
On a second thought: Why do you need this factory at all? You could also add the `HttpClientFactory` to the handler factory and pass the http client to handler during initialization.

Reply:
Sorry, I fail with this change. As soon as I add the HttpClientFactory to the SenecHomeHandlerFactory my code doesn't compile any more and I don't understand why.

Reply:
What error does it give?

Reply:
For me it is unclear how it works right now, who calls `public SenecHomeApiFactory(@Reference HttpClientFactory httpClientFactory)`? Where is that `httpClientFactory` created? Once I move the object I don't know how to handle it correctly. A code example or change proposal would help.

Reply:
`SenecHomeApiFactory` and `HttpClientFactory` have both `@Component` annotation. This will trigger the OSGI framework to create these classes and inject possible arguments. That are guided by the `@Reference` and `@Activate` annotations. See my other comment on the suggested changes.